etcdserver/server.go

```diff
@@ -521,9 +523,54 @@ func NewServer(cfg ServerConfig) (srv *EtcdServer, err error) {
 	}
 	srv.r.transport = tr
 
+	activePeers := 0
+	for _, m := range cl.Members() {
+		if m.ID != id {
+			if tr.IsActive(m.ID) {
+				activePeers++
+			}
+		}
+	}
+
+	clusterN := len(cl.Members())
+	plog.Infof("%s has %d active peers among %d found member(s)", srv.ID(), activePeers, clusterN)
+
+	// only advance ticks iif
+	//   1. starting to fresh cluster (restart == false)
+	//   AND
+	//   2-1. single-node cluster (no peer to wait for)
+	//   OR
+	//   2-2. no active peer connection has been established
+	//        (e.g. node that starts very first out of 3-node)
+	if !restart && (clusterN == 1 || activePeers == 0) {
+		// save one tick; leader may heartbeat before timeout
+		tick := cfg.ElectionTicks - 1
+		plog.Infof("%s is advancing %d ticks for faster election (election tick %d)", srv.ID(), tick, cfg.ElectionTicks)
+		advanceTicksForElection(n, tick)
+	} else {
+		// on restart, there is likely an active peer already
```

Comment:
even for restart case, we should still consider the number of active member. if there is none, we still can advance ticks, no?

Reply:
Yea, we would need wait until the local node finds its peers (`cl.Members() > 0`), to do that. I will play around it to address https://github.com/coreos/etcd/pull/9364/files#r171727961.